### PR TITLE
Bitbucket OAuth. Resolve #201.

### LINKED ISSUE #201: bitbucket OAuth
Should probably use OAuth as well.

### PR DESCRIPTION
**Pull request by ryneeverett**
- Implement OAuth2. Thanks to @ibuchanan for helping me navigate
  bitbucket's idiosyncratic implementation.
- Add a bugwarrior.data file to the ~/.task directory, per [taskwarrior's
  third-party guidelines](https://taskwarrior.org/docs/3rd-party.html#guidelines).

### CODE REVIEW

**Comment by ryneeverett** [comment]
Where are you Travis?

**Comment by ralphbean** [comment]
The oauth bits look fine to me.  Can you look at the lockfile question above and rebase on ``develop``?

File: bugwarrior/data.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+def get(key):
+    try:
+        with open(DATAFILE, 'r') as jsondata:
+            data = json.load(jsondata)
+            return data[key]
+    except IOError:  # File does not exist.
+        return None
+
+
+def set(key, value):
```
Comment (ryneeverett) [inline bugwarrior/data.py]: Yes, good catch.

**Comment by ryneeverett** [comment]
I still gotta look at the lockfile issue.

FYI, I [tested out](https://github.com/ryneeverett/bugwarrior/tree/bitbucket-oauth2-dbm) using [`dbm`](https://docs.python.org/3/library/dbm.html) as a key/value store since we wouldn't have to repetitiously serialize, deserialize and overwrite the whole file. It's beautifully simple, but on some systems it creates additional `.dir` and `.pag` files and that doesn't seem acceptable.

**Comment by ryneeverett** [comment]
FYI, I did add a lockfile.

**Comment by ralphbean** [comment]
Oh, I'd missed it.  :+1: here.  Feel free to merge.  :dango:

**Pull request merged**